### PR TITLE
[Home] Set rails to null during network failure

### PR DESCRIPTION
**Pull request by mennenia**
For #333 

I personally think this is even worse than infinite spinners. 

With infinite spinners the user might think something got stuck and will kill the app to try and reload. Now, it's empty, and they may have no clue what to do about that or understand that a force restart of the app would help. It just looks like the app decided to be broken, with no indication why. 

This is just my view... I completely understand the step-by-step approach and that we ideally want it to just refresh and work. 

simulator screen shot 19 oct 2016 10 54 49

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">No CHANGELOG added.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">The following files do not have tests: <a href="https://github.com/artsy/emission/blob/935ec871f5d5154cb56333c8c8a7dc4bf6f4d3ba/lib/app_registry.js">lib/app_registry.js</a>, <a href="https://github.com/artsy/emission/blob/935ec871f5d5154cb56333c8c8a7dc4bf6f4d3ba/lib/components/artwork_grids/infinite_scroll_grid.js">lib/components/artwork_grids/infinite_scroll_grid.js</a>, <a href="https://github.com/artsy/emission/blob/935ec871f5d5154cb56333c8c8a7dc4bf6f4d3ba/lib/components/buttons/flat_white.js">lib/components/buttons/flat_white.js</a>, <a href="https://github.com/artsy/emission/blob/935ec871f5d5154cb56333c8c8a7dc4bf6f4d3ba/lib/components/gene/artworks.js">lib/components/gene/artworks.js</a>, <a href="https://github.com/artsy/emission/blob/935ec871f5d5154cb56333c8c8a7dc4bf6f4d3ba/lib/components/gene/header.js">lib/components/gene/header.js</a>, <a href="https://github.com/artsy/emission/blob/935ec871f5d5154cb56333c8c8a7dc4bf6f4d3ba/lib/components/home/artist_rails/artist_rail.js">lib/components/home/artist_rails/artist_rail.js</a>, <a href="https://github.com/artsy/emission/blob/935ec871f5d5154cb56333c8c8a7dc4bf6f4d3ba/lib/components/home/artwork_rails/artwork_rail.js">lib/components/home/artwork_rails/artwork_rail.js</a>, <a href="https://github.com/artsy/emission/blob/935ec871f5d5154cb56333c8c8a7dc4bf6f4d3ba/lib/relay/config.js">lib/relay/config.js</a> &amp; <a href="https://github.com/artsy/emission/blob/935ec871f5d5154cb56333c8c8a7dc4bf6f4d3ba/lib/relay/routes.js">lib/relay/routes.js</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by alloy** [comment]
Yeah, this is definitely worse, because what I would like the best worst-case scenario to look like is that the rail is completely taken out of the list, i.e. no empty space is left.

**Comment by mennenia** [comment]
I'm confused :o.

This doesn't create empty space - there's nothing there. 

I looked into seeing if I could unmount the component, to see if that meant that from the home container, we could see if components were taken out and hence put them back in, but from what I have tried that doesn't seem possible.

**Comment by alloy** [comment]
> This doesn't create empty space - there's nothing there.

Oohh, I see, you mean that _all_ rails got rendered to _null_ ?

I’m thinking of the more typical case where a single rail fails, because If _all_ rails would fail to load then I think it’s reasonable to assume that the initial root home query would also fail, in which case a ‘FAILED TO LOAD’ view would be presented.

File: lib/components/home/artist_rails/artist_rail.js [hunk 1]
```diff
@@ -152,6 +157,10 @@ class ArtistRail extends React.Component {
   }
 
   render() {
+    if (this.state.loadFailed !== null && this.state.loadFailed) {
```
Comment (mennenia) [inline lib/components/home/artist_rails/artist_rail.js]: Pushed :)

File: lib/components/home/artist_rails/artist_rail.js [hunk 2]
```diff
@@ -29,12 +29,17 @@ class ArtistRail extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-      artists: []
+      artists: [],
+      loadFailed: null,
```
Comment (alloy) [inline lib/components/home/artist_rails/artist_rail.js]: I’m thinking now that maybe this should just be `false` initially and you should also add this variable to the list of `state` types above (at line 27).

Comment (mennenia) [inline lib/components/home/artist_rails/artist_rail.js]: ok pushed properly now. Sorry for all the commits >_<

**Comment by alloy** [comment]
Thanks!

**Pull request merged**